### PR TITLE
Add .gitignore support

### PR DESCRIPTION
**Pull request by bjorns**
Read .gitignore files in the repo and use them to skip ignored files from upload.

Flag `--no-gitignore` disables the functionality.

### CODE REVIEW

**Comment by iamjackg** [comment]
Thanks for the PR! I fully support this and I think it's a great idea, but I have a few suggestions:

* rather add the parser as a dependency rather than straight up including it. Would you mind reworking your MR to do that instead? 
* I would also like for this to be optionally turned off through a `-I` or `--no-ignore` flag from the command line
* while I appreciate your use of `logging` (and it's definitely something I want to clean up in the rest of the codebase) I'd like to keep things uniform for now and limit output to simple `print`s.

It would be nice to also add a test or two for this :)

**Comment by bjorns** [comment]
Alright, so I've added a test and a --no-gitignore flag. I didn't add the `-I` flag because none of the other directory flags had a short form. Maybe that's something you want to add in a concerted way instead?

Also unit test added.

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[pytest]
```
Comment (bjorns) [inline pytest.ini:1]: The gitignore_parser package triggers a warning that is exposed in purest. This ignores the warning. I have found no functional effect of the warning.

**Comment by iamjackg** [comment]
Thank you so much for working on this, I'll take a look soon.

File: md2cf/confluence_renderer.py [hunk 2]
```diff
@@ -82,6 +82,15 @@ def plain_text_body(self, text):
         body_tag.text = text
         return body_tag
 
+    def text(self, text):
```
Comment (iamjackg) [inline md2cf/confluence_renderer.py:85]: Looks like you accidentally kept this from your other PR.

File: md2cf/document.py [hunk 3]
```diff
@@ -38,11 +39,14 @@ def get_content_hash(self):
...
 
 def find_non_empty_parent_path(
-    current_dir: Path, folder_data: Dict[Path, Dict[str, Any]]
+    current_dir: Path,
+    folder_data: Dict[Path, Dict[str, Any]],
+    default: Path
 ) -> Path:
     for parent in current_dir.parents:
-        if folder_data[parent]["n_files"]:
+        if parent in folder_data and folder_data[parent]["n_files"]:
```
[the remark this answers is not in the document]
Comment (iamjackg) [reply md2cf/document.py:47]: Ah interesting, I'll play around with it a bit and see if there was an edge case I missed. Thanks!

File: md2cf/confluence_renderer.py (hunk 2 above)
Comment (iamjackg) [reply md2cf/confluence_renderer.py:85]: I'll merge it and remove it myself :)

**Pull request merged**